### PR TITLE
Cleaning up README and moving documentation to docs

### PR DESCRIPTION
**Pull request by jodobear**
House cleaning for README and initiating structured docs.

TO DO:
* Add fidelity bond doc
* Update project Roadmap and TODO

### CODE REVIEW

**Review by rajarshimaitra — COMMENTED** [review]
ACK a3e5477d1566958cc71a373c3ec25c03677455f7

The idea looks good. Maybe we can have a more clear name for the `how_to` doc?

File: README.md [hunk 1]
```diff
@@ -15,395 +15,18 @@ Project design document: [Design for a CoinSwap Implementation for Massively Imp
...
-* [Waxwing's blog post from 2017 about CoinSwap](https://web.archive.org/web/20200524041008/https://joinmarket.me/blog/blog/coinswaps/)
-
-* [gmaxwell's original coinswap writeup from 2013](https://bitcointalk.org/index.php?topic=321228.0). It explains how CoinSwap actually works. If you already understand how Lightning payment channels work then CoinSwap is similar.
-
-* [Design for improving JoinMarket's resistance to sybil attacks using fidelity bonds](https://gist.github.com/chris-belcher/18ea0e6acdb885a2bfbdee43dcd6b5af/). Document explaining the concept of fidelity bonds and how they provide resistance against sybil attacks.
-
+## Docs
+Check [how_to.md](docs/how_to.md) for steps on how to create a coinswap with this implementation.
 
+Check [Developer Resources](docs/developer_resources.md) on information on the protocol and further reading.
```
Comment (rajarshimaitra) [inline README.md:30]: Can we add build and test instructions here too? Just two lines of build the binary with `cargo build`, and run all the tests with `cargo test`. Its trivial, but just not written anywhere.

File: README.md [hunk 2]
```diff
@@ -15,395 +15,18 @@ Project design document: [Design for a CoinSwap Implementation for Massively Imp
...
-    - [Protocol between takers and makers](#protocol-between-takers-and-makers)
-    - [Code Structure](#code-structure)
-    - [Further reading](#further-reading)
+  - [Docs](#docs)
   - [Chris Belcher's personal roadmap for the project](#chris-belchers-personal-roadmap-for-the-project)
   - [Community](#community)
 
 ## State of the project
 
 The project is nearly usable, though it doesnt have all the necessary features yet. The code written so far is published for developers and power users to play around with. It doesn't have config files yet so you have to edit the source files to configure stuff. It is possible to run it on mainnet, but only the brave will attempt that, and only with small amounts.
```
Comment (rajarshimaitra) [inline README.md:24]: Maybe we can add some further information on what this library is currently. Along the line of, it's a cli app written in rust as demo prototype of working with underlying coinswap subroutines and primitives.

File: docs/how_to.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
+# Documentation on Creating Coinswap
```
Comment (rajarshimaitra) [inline docs/how_to.md:1]: Maybe we can call this an "Example Coinswap with `teleport` cli app". 

We might not have this binary later. We will make sure to update this example accordingly.

File: README.md (hunk 1 above)
Comment (jodobear) [reply README.md:47]: Yep, added an "Install/Build from Source" section in the README and removed it from `app_instructions`

**Comment by jodobear** [comment]
> ACK [a3e5477](https://github.com/utxo-teleport/teleport-transactions/commit/a3e5477d1566958cc71a373c3ec25c03677455f7)
> 
> The idea looks good. Maybe we can have a more clear name for the `how_to` doc?

changed it to `app_instructions`

**Review by rajarshimaitra — APPROVED** [review]
ACK 13b166e41e266588ef0be67ee70e383b23f5013c

**Pull request merged**